splicemachine/spliceengine
# DB-12106 Fix NPE in MaxMinAggregator

## Description

`MaxMinAggregator.toProtobufBuilder` used to call `value.toProtobuf()` without checking whether `value` is null or not. 
This PR makes `MaxMinAggregator.toProbufBuilder` behave like `SumAggregator.toProtobufBuilder` and checks for `value != null`

## How to test

```
drop table t0;

drop table t2;

CREATE TABLE t0(c0 boolean , c1 TIMESTAMP);

CREATE TABLE t2(c0 TIMESTAMP not null);

INSERT INTO T0(C0) VALUES (FALSE);

INSERT INTO T0(C0, C1) VALUES (TRUE, '1970-01-13 04:08:02'), (TRUE, '1969-12-12 03:20:09');

INSERT INTO T0(C1) VALUES ('1969-12-12 03:20:09');

INSERT INTO T0(C1) VALUES ('1970-01-13 08:53:47');

SELECT MAX(agg0) FROM (SELECT MAX(T2.C0)  as agg0 FROM T0 LEFT OUTER JOIN T2 ON ((T0.C1)!=(T2.C0)) WHERE T0.C0 UNION ALL SELECT MAX(T2.C0)  as agg0 FROM T0 LEFT OUTER JOIN T2 ON ((T0.C1)!=(T2.C0)) WHERE NOT (T0.C0) UNION ALL SELECT MAX(T2.C0)  as agg0 FROM T0 LEFT OUTER JOIN T2 ON ((T0.C1)!=(T2.C0)) WHERE (T0.C0) IS NULL) as asdf;
```
should not throw an NPE anymore
